test/e2e/diagram.overlays.rendering.test.ts

```diff
@@ -115,38 +159,75 @@ describe('BPMN elements with overlays', () => {
       customSnapshotIdentifier: 'remove.all.overlays.of.shape',
     });
   });
+});
 
-  // TODO tests will be added with https://github.com/process-analytics/bpmn-visualization-js/issues/1166
-  it.skip.each([<OverlayEdgePosition>'start', <OverlayEdgePosition>'middle', <OverlayEdgePosition>'end'])(
-    `add overlay on SequenceFlow on %s`,
-    async (position: OverlayEdgePosition) => {
+describe('BPMN Edges with overlays', () => {
+  const bpmnDiagramNameAssociations = 'overlays.edges.associations.complex.paths';
+  const bpmnDiagramNameMessageFlows = 'overlays.edges.message.flows.complex.paths';
+  const bpmnDiagramNameSequenceFlows = 'overlays.edges.sequence.flows.complex.paths';
+  function getEdgeKindAndIds(bpmnDiagramName: string): [string, string[]] {
+    let edgeKind: string;
+    let ids: string[];
+    switch (bpmnDiagramName) {
+      case bpmnDiagramNameAssociations:
+        edgeKind = 'association';
+        ids = ['Association_1opueuo', 'Association_0n43f9f', 'Association_01t0kyz'];
+        break;
+      case bpmnDiagramNameSequenceFlows:
+        edgeKind = 'sequence';
+        ids = ['Flow_039xs1c', 'Flow_0m2ldux', 'Flow_1r3oti3', 'Flow_1byeukq'];
+        break;
+      case bpmnDiagramNameMessageFlows:
+        edgeKind = 'message';
+        ids = [
+          // incoming and outgoing flows of the 2 pools starting from the right
+          'Flow_0skfnol',
+          'Flow_0ssridu',
+          'Flow_0s4cl7e',
+          'Flow_0zz7yh1',
+          // flows in the middle of the diagram
+          'Flow_0vsaa9d',
+          'Flow_17olevz',
+          'Flow_0qhtw2k',
+          // flows on the right
+          'Flow_0mmisr0',
+          'Flow_1l8ze06',
+        ];
+        break;
+    }
+    return [edgeKind, ids];
+  }
+  describe.each([bpmnDiagramNameSequenceFlows, bpmnDiagramNameAssociations, bpmnDiagramNameMessageFlows])('diagram %s', (bpmnDiagramName: string) => {
```

Comment:
Or directly give the diagram file + the edge kind + the edges id as an array of array. It's more readable to know what we test.

Reply:
I choose this solution, it is clearer. See 91d206b